plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,375 @@
+
+            if l3out_node_routing_policy_object and mso.existing.get("nodeRoutingPolicyRef") != l3out_node_routing_policy_object.details.get("uuid"):
+                ops.append(
+                    dict(op="replace", path=node_group_policy_path + "/nodeRoutingPolicyRef", value=l3out_node_routing_policy_object.details.get("uuid"))
+                )
+                proposed_payload["nodeRoutingPolicyRef"] = l3out_node_routing_policy_object.details.get("uuid")
+
+            elif node_routing_policy == "" and mso.existing.get(
+                "nodeRoutingPolicyRef"
+            ):  # Clear the node routing policy when node_routing_policy is empty string
+                ops.append(dict(op="replace", path=node_group_policy_path + "/nodeRoutingPolicyRef", value=node_routing_policy))
+                proposed_payload["nodeRoutingPolicyRef"] = node_routing_policy
+
+            if bfd:
+                if bfd.get("state") == "disabled" and proposed_payload.get("bfdMultiHop"):
+                    proposed_payload.pop("bfdMultiHop")
+                    ops.append(dict(op="remove", path=node_group_policy_path + "/bfdMultiHop"))
+                else:
+                    if not proposed_payload.get("bfdMultiHop"):
+                        proposed_payload["bfdMultiHop"] = dict()
+                        ops.append(dict(op="replace", path=node_group_policy_path + "/bfdMultiHop", value=dict()))
+
+                    if bfd.get("auth") and mso.existing.get("bfdMultiHop", {}).get("authEnabled") is not (True if bfd.get("auth") == "enabled" else False):
+                        ops.append(
+                            dict(
+                                op="replace",
+                                path=node_group_policy_path + "/bfdMultiHop/authEnabled",
+                                value=True if bfd.get("auth") == "enabled" else False,
+                            )
+                        )
+                        proposed_payload["bfdMultiHop"]["authEnabled"] = True if bfd.get("auth") == "enabled" else False
+
+                    if bfd.get("key_id") is not None and mso.existing.get("bfdMultiHop", {}).get("keyID") != bfd.get("key_id"):
+                        ops.append(dict(op="replace", path=node_group_policy_path + "/bfdMultiHop/keyID", value=bfd.get("key_id")))
+                        proposed_payload["bfdMultiHop"]["keyID"] = bfd.get("key_id")
+
+                    if bfd.get("key") is not None:
+                        if not isinstance(proposed_payload["bfdMultiHop"].get("key"), dict):
+                            ops.append(dict(op="replace", path=node_group_policy_path + "/bfdMultiHop/key", value=dict()))
+
+                        ops.append(dict(op="replace", path=node_group_policy_path + "/bfdMultiHop/key/value", value=bfd.get("key")))
+                        proposed_payload["bfdMultiHop"]["key"] = dict(value=bfd.get("key"))
+
+            if target_dscp is not None and mso.existing.get("targetDscp") != target_dscp:
```

Comment:
target_dscp is either a None value or a choice which is a none empty string, so none check is not required and we could do falsified check
```suggestion
            if target_dscp and mso.existing.get("targetDscp") != target_dscp:
```